LanguageFeatures/Inline-classes/static_analysis_inline_class_A17_t05.dart

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion A compile-time error occurs if an extension type has a
+/// non-extension superinterface whose transitive alias expansion is a type
+/// variable, a deferred type, any top type (including dynamic and void), the
+/// type Null, any function type, the type Function, any record type, the type
+/// Record, or any type of the form T? or FutureOr<T> for any type T.
+///
+/// @description Checks that it is a compile-time error if an extension type
+/// has a non-extension superinterface whose transitive alias expansion is a
+/// function type
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=inline-class
+
+typedef void Foo1();
+typedef Foo2 = int Function(int);
```

Comment:
It would also be nice to have a test where the type following the `implements` keyword is a function type directly, not a type alias that expands into a function type.

Reply:
Added